File: src/BooksReader.Core/Models/DTO/Public/BookMarketDto.cs
using System;
using System.Collections.Generic;
using System.Text;
using BooksReader.Core.Entities;
using BooksReader.Core.Enums;
using BooksReader.Core.Infrastrcture;

namespace BooksReader.Core.Models.DTO.Public
{
    public class BookMarketDto : GeneralBook, ISemanticUrl
    {
        public string SemanticUrl { get; set; }
        public string AuthorSemanticUrl { get; set; }

        public Guid AuthorId { get; set; }
        public DateTimeOffset? Published { get; set; }
        public int SubscriptionDurationDays { get; set; }

        // not sure that it should be shown at market page
        public SubscriptionStatus Subscription { get; set; }

        public IEnumerable<BookPrice> BookPrices { get; set; }
    }
}

File: src/BooksReader.Web/ClientApp/src/app/modules/core/models/api/dto/public/book-market.dto.ts
import { BookPrice } from '../../entities';
import { SubscriptionStatus } from '@br/core/enums/subscription-status';
import { SiteConstants } from '@br/config';

export interface BookMarketDto {
    bookId: string;
    semanticUrl?: string;
    title: string;
    author: string;
    authorId: string;
    authorSemanticUrl?: string; 
    description: string;
    picture: string;
    isForSale: boolean;
    published: Date;
    subscriptionDurationDays?: number


    bookPrices: BookPrice[];
    subscription: SubscriptionStatus;
}

export const EmptyBookMarketDto: BookMarketDto = {
    bookId: '',
    author:'',
    authorId: '',
    description:'',
    title:'',
    isForSale: false,
    picture: SiteConstants.defaultImage,
    published: null,
    subscription: SubscriptionStatus.none,
    bookPrices: []
};

